library/ssl_cli.c

```diff
@@ -3411,27 +3406,22 @@ static int ssl_write_client_key_exchange( mbedtls_ssl_context *ssl )
         if( status != PSA_SUCCESS )
             return( MBEDTLS_ERR_SSL_HW_ACCEL_FAILED );
 
-        /* Export the public part of the ECDH private key from PSA
-         * and convert it to ECPoint format used in ClientKeyExchange. */
+        /* Export the public part of the ECDH private key from PSA.
+         * The export format is an ECPoint structre as expected by TLS,
+         * but we just need to add a length byte before that. */
+        unsigned char *own_pubkey = ssl->out_msg + header_len + 1;
+        unsigned char *end = ssl->out_msg + MBEDTLS_SSL_OUT_CONTENT_LEN;
```

Comment:
Is it OK to change the maximal `pubkey` length from `MBEDTLS_PSA_MAX_EC_PUBKEY_LENGTH` to 

> `MBEDTLS_SSL_OUT_CONTENT_LEN - header_len - 1`? 

Maybe it should be the smaller of two?

Reply:
Good question, but I think it's OK. The previous limitation was only because we used a temporary stack buffer, it's not something we were trying to enforce for security or compliance reasons, it was just because in practice we had a fixed-size buffer that we didn't want to overflow.

Also, in practice the `pubkey` output will always be `<= MBEDTLS_PSA_MAX_EC_PUBKEY_LENGTH` unless the PSA implementation is buggy. I think it's OK generally for the TLS layer to assume that the PSA implementation is correct.